src/vault/auth/userpass.clj

```diff
@@ -0,0 +1,36 @@
+(ns vault.auth.userpass
+  "The /auth/userpass endpoint manages username & password authentication
+  functionality.
+
+  Reference: https://www.vaultproject.io/api-docs/auth/userpass"
+  (:require
+    [vault.client.http :as http]
+    [vault.client.proto :as proto]
+    [vault.util :as u])
+  (:import
+    vault.client.http.HTTPClient))
+
+
+(defprotocol API
+  "The userpass endpoints manage username & password authentication
+  functionality."
+
+  (login
+    [client username password]
+    "Login with the username and password."))
```

Comment:
It'd be helpful to describe both the returned data as well as the side-effect that this updates the client auth.

Reply:
I took a crack at describing it but am happy to change further if you think more specifics are warranted or just better wording.